toolchain/lex/lex.cpp

```diff
@@ -94,21 +97,38 @@ class [[clang::internal_linkage]] Lexer {
 
   auto current_line() -> LineIndex { return LineIndex(line_index_); }
 
-  auto current_line_info() -> TokenizedBuffer::LineInfo* {
+  auto current_line_info() -> LineInfo* {
     return &buffer_.line_infos_[line_index_];
   }
 
   auto next_line() -> LineIndex { return LineIndex(line_index_ + 1); }
 
-  auto next_line_info() -> TokenizedBuffer::LineInfo* {
+  auto next_line_info() -> LineInfo* {
     CARBON_CHECK(line_index_ + 1 <
                  static_cast<ssize_t>(buffer_.line_infos_.size()));
     return &buffer_.line_infos_[line_index_ + 1];
   }
 
-  auto NoteWhitespace() -> void {
-    buffer_.token_infos_.back().has_trailing_space = true;
-  }
+  // Note when the lexer has encountered whitespace, and the next lexed token
+  // should reflect that it was preceded by some amount of whitespace.
+  auto NoteWhitespace() -> void;
+
+  // Add a lexed token to the tokenized buffer, and reset any token-specific
+  // state tracked in the lexer for the next token.
+  auto AddLexedToken(TokenInfo info) -> TokenIndex;
+
+  // An overload for lexing a token.Builds the correctly encoded token info,
+  // adds it to the tokenized buffer and returns the token index.
+  //
+  // This overload is for tokens without a payload.
```

Comment:
```suggestion
  // Lexes a token with no payload: builds the correctly encoded token info,
  // adds it to the tokenized buffer and returns the token index.
```

Per the style guide, function comments should start with a verb phrase with "this function" as the implied subject. It also seems helpful (and more concise) to frontload the payload aspect a little more, but I don't feel strongly about that part.